### PR TITLE
docs: Add page describing how to use events

### LINKED ISSUE #678: [REQUEST] Adding a section describing how Client event can be used
### Describe the feature.

I think since the library itself does support Discord gateway events, there should be a dedicated section for examples about how these events can be used in production, such as making a simple example about ``on_message_create`` or things like that.

### Code of Conduct

- [X] I agree to follow the contribution requirements.

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This pull request is about (X), which does (Y).

## Checklist

- [ ] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [x] Documentation
  - [ ] Breaking change
  - [ ] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

**Review by B1ue-Dev — APPROVED** [review]
Approved!

File: docs/events.rst [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+After this, lets look at events from the Discord API.
+
+Discord API Events
+******************
+
+Events in this section do not match the name needed to put into the function. Please check
+:ref:`above <events:How to listen for events>` for how to get the correct name.
+
+
+There is a lot of events dispatched by the Discord API. All of those can be found `here`_.
```
Comment (Toricane) [inline docs/events.rst]:
```suggestion
There are a lot of events dispatched by the Discord API. All of those can be found `here`_.
```

File: docs/events.rst [hunk 2]
```diff
@@ -0,0 +1,180 @@
+Event Documentation
+====================
+
+You all probably already know that there are several events, internal and from discord, that you can receive with your
+bot. This page will lead you through all dispatched internal events and a few from discord.
```
Comment (Toricane) [inline docs/events.rst]:
```suggestion
You all probably already know that there are several events, internal and from Discord, that you can receive with your
bot. This page will lead you through all dispatched internal events and a few from Discord.
```

File: docs/events.rst [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+There are several different internal events:
+
+    - ``raw_socket_create``
+    - ``on_interaction``
+    - ``on_command``
+    - ``on_component``
+    - ``on_autocomplete``
+    - ``on_modal``
+
+Lets now have a look on those events in detail:
```
Comment (Toricane) [inline docs/events.rst]:
```suggestion
Let's now have a look on those events in detail:
```

File: docs/events.rst [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+The events ``HELLO``, ``RESUMED``, ``RECONNECT``, ``INVALID SESSION`` and ``TYPING START`` are not dispatched by the library.
+
+``TYPING START`` will be included into the :raw socket create event. You can
+also listen for it if you choose to subclass the WebSocketClient
+
+The event ``VOICE STATE UPDATE`` can be only received with the voice :ref:`Extension <faq:Extension Libraries>`.
+
+
+Lets now have a look at a few events:
```
Comment (Toricane) [inline docs/events.rst]:
```suggestion
Let's now have a look at a few events:
```

File: docs/events.rst [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+Event: ``READY``
+^^^^^^^^^^^^^^^^
+This event fires whenever ``READY`` is dispatched by discord. This happens when connecting to the web socket server.
```
Comment (Toricane) [inline docs/events.rst]:
```suggestion
This event fires whenever ``READY`` is dispatched by Discord. This happens when connecting to the web socket server.
```

**Review by Toricane — DISMISSED** [review]
Grammar suggestions
Pls also replace any "discord" with "Discord" if i missed any

**Review by i0bs — APPROVED** [review]

**Comment by i0bs** [comment]
Marking Toricane's review for dismissal as stale content

**Pull request merged**